sovryn-token-bridge/bridge/contracts/Bridge.sol

```diff
@@ -207,14 +212,18 @@ contract Bridge is Initializable, IBridge, IERC777Recipient, UpgradablePausable,
         //// Bridge v4 upgrade functions
         if (tokenAddress == WETHAddr) {
             address payable payableReceiver = address(uint160(receiver));
-            (bool success, ) =
-                payableReceiver.call.value(amount)("");
-            require(success, "Bridge: Failed to send ETH to receiver");
+            if (isBridgeReceiver[receiver]) {
+                IBridgeReceiver(payableReceiver).receiveEthFromBridge.value(amount)(userData);
+            } else {
+                (bool success, ) =
+                    payableReceiver.call.value(amount)("");
```

Comment:
if you provide `userData` here instead of empty string, then one can use it to initiate a meaningful interaction with any contract on the receiving side of the bridge. This might or might not be desirable...

Reply:
This is true.
The bridge already enables interaction with SC on the other side with no limit (with any transfer on the bridge that uses extraData).
It enables deposit also to an exchange address or to a reliable SC, but can be used also with not desirable SC.

Reply:
in general, evgeny is right. we could add it it to any contract call. however, this might break compatability with contracts which only implement the receive function and not the fallback function. i think this requires further internal discussion, but for now, the important things is to allow for the bridge interaction.